budgetportal/tests/test_irm_preprocessor.py

```diff
@@ -110,3 +110,21 @@ def test_get_row_implementors_just_one(self):
         self.assertEqual(implementors[1], "")
         self.assertEqual(implementors[2], "C")
         self.assertEqual(implementors[3], "")
+
+
+class PreprocessEmptyRowTestCase(unittest.TestCase):
+    def test_empty_rows_removed(self):
+        headers = BASE_HEADERS + [REPEATED_IMPLEMENTOR_HEADER] * 20
+        dataset = Dataset(headers=headers)
+        num_nonempty_rows = dataset.height
```

Comment:
the dataset doesn't have any nonempty rows at this point, does it?

Please add a nonempty row or two

Reply:
One non-empty row added.

https://github.com/vulekamali/datamanager/pull/213/commits/4b7fb0d949828d11deb02493dde613d4de3f13db